test/test_autograd.py

```diff
@@ -3346,7 +3350,7 @@ class DeepReentrant(Function):
             @staticmethod
             def forward(ctx, x):
                 with torch.enable_grad():
-                    ctx.x = Variable(x.data, requires_grad=True)
+                    ctx.x = Variable(x, requires_grad=True)
```

Comment:
You want a `.detach()` here I think. As you don't want `ctx.x` to be linked to `x` in a differentiable manner.